src/mpid/ch4/shm/posix/release_gather/nb_reduce_release_gather.h

```diff
@@ -0,0 +1,511 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef NB_REDUCE_RELEASE_GATHER_H_INCLUDED
+#define NB_REDUCE_RELEASE_GATHER_H_INCLUDED
+
+#include "gentran_types.h"
+
+#define MPIDI_POSIX_RELEASE_GATHER_NB_IREDUCE_GATHER_FLAG_ADDR(rank, segment, num_ranks)        \
+    (((MPL_atomic_uint64_t *)nb_release_gather_info_ptr->ireduce_flags_addr) +                     \
```

Comment:
The cast to `MPL_atomic_uint64_t *` here gives me some concern. I'm not sure what the intent is, but it won't cause an atomic read.

Reply:
This macro only returns the address, which is used later in calls e.g. MPL_atomic_acquire_load_uint64 in line 70